### PR TITLE
Library structure refactor

### PR DESCRIPTION
**Pull request by Janix4000**
Propose new code architecture, which can be found in `testing` directory

### CODE REVIEW

**Comment by Janix4000** [comment]
[Create global constants describing library structure](https://trello.com/c/gHhPDtIv/37-create-global-constants-describing-library-structure)

File: votepy/testing/algorithms/base_algorithm.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from votepy.ordinal_election import OrdinalElection
```
Comment (Wolokin) [inline votepy/testing/algorithms/base_algorithm.py]: Should all these files reside under testing?

File: votepy/testing/rules/chamberlin_courant.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                if candidate in committee:
+                    score += max(number_of_scored_candidates - i, 0)
+                    break
+        return score
+
+    brute_force.prepare(scoring_function)
+    return brute_force.solve(voting, size_of_committee)
+
+
+@impl(chamberlin_courant, Greedy)
```
Comment (Wolokin) [inline votepy/testing/rules/chamberlin_courant.py]: I think we should be consistent here (with i.e. line 27), either use keywords everywhere, or nowhere

**Review by Wolokin — APPROVED** [review]
Looks good, I like it. Just please address these two small comments

File: votepy/testing/rules/chamberlin_courant.py (hunk 2 above)
Comment (Janix4000) [reply votepy/testing/rules/chamberlin_courant.py]: I think this version without keywords is self-explanatory enough, I will change line 27.

File: votepy/testing/algorithms/base_algorithm.py (hunk 1 above)
Comment (Janix4000) [reply votepy/testing/algorithms/base_algorithm.py]: No, I will move them one level up in the files' hierarchy and update imports

File: votepy/testing/solve.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from typing import Callable, Union
+from votepy.testing.algorithms.base_algorithm import BaseAlgorithm
+from votepy.testing.structure.structure import get_algorithm, get_implementation
+
+
+def solve(rule: Union[Callable, str], *rule_args, algorithm: BaseAlgorithm, **rule_kwargs):
```
Comment (Janix4000) [inline votepy/testing/solve.py]: Add docstring.

File: votepy/testing/structure/structure.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    >>> @algo(name='name')
+    ... class Algo(BaseAlgorithm):
+    ...     def _solve():
+    ...         pass
+    >>>
+        get_algorithm('name') == Algo()
+    True
+        get_algorithm(Algo) == Algo()
+    True
+        get_algorithm(Algo()) == Algo()
```
Comment (Janix4000) [inline votepy/testing/structure/structure.py]: Add example algorithm with constructor parameters.

**Review by Janix4000 — COMMENTED** [review]
Improve functions' documentation.

**Review by msluszniak — APPROVED** [review]

**Review by Wolokin — APPROVED** [review]
APPROVED

**Review by msluszniak — APPROVED** [review]

**Pull request merged**